### PR TITLE
Fix cypress.config.ts problem

### PR DESCRIPTION
**Pull request by DavidFontalba**
We've encountered different problems using the plugin in our Cypress project using TypeScript, first of all, `runFailed.js` didn't find our Cypress configuration and when we fixed that, the "Cannot use import statement outside a module" error appeared. The problem is that we need the config before initializing Cypress, so we solved that by removing that line. Now the plugin' script is forced to be in the correct directory when executing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,17 +91,17 @@ module.exports = defineConfig({
 npx cypress run
 ```
 
-2. If there are failed tests, run the following command from the **directory of the project's `cypress.config`**:
+2. If there are failed tests, run the following command from the **failedTestDirectory of the project**:
```
Comment (dennisbergevin) [inline README.md:94]: The `failedTestDirectory` change here assumes the user will be customizing the path for storing failed tests. I'd rather keep it as an option within the plugin to set the environment variable `failedTestDirectory`.

File: README.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ For convenience, you may desire to house the `npx` command within an npm script
 
 ```json
   "scripts": {
-    "last-failed": "npx cypress-run-last-failed run --e2e --browser electron"
+    "last-failed": "cd ${failedTestDirectory} && npx cypress-plugin-last-failed run --e2e --browser electron"
```
Comment (dennisbergevin) [inline README.md:144]: Could you help me understand how the script here will read and understand `${failedTestDirectory}`?

**Comment by dennisbergevin** [comment]
Hey @DavidFontalba! Thank you for opening a PR and addressing some issues you've encountered.

I left a few comments to understand a bit more on what your proposed changes are.

File: README.md [hunk 3]
```diff
@@ -91,17 +91,17 @@ module.exports = defineConfig({
 npx cypress run
 ```
 
-2. If there are failed tests, run the following command from the **directory of the project's `cypress.config`**:
+2. If there are failed tests, run the following command from the **failedTestDirectory of the project**:
 
 ```bash
-npx cypress-last-failed run
+npx cypress-plugin-last-failed run
```
[the remark this answers is not in the document]
Comment (DavidFontalba) [reply README.md:97]: The name change was because when i execute it like in the README I got this npm error and only with the updated name it's solved
imagen

File: README.md (hunk 2 above)
Comment (DavidFontalba) [reply README.md:144]: It does not pretend to the script to understand it as a variable; it should be settled up by the user with the correct directory, which is the same as you used as `failedTestDirectory`. How do you think we could indicate it better?

File: runFailed.js [hunk 4]
```diff
@@ -4,18 +4,14 @@ const cypress = require('cypress');
...
   const noFailedTestsMessage =
     'No previous failed tests detected\nTry running tests again with cypress run';
 
   // Use the cypress environment variable failedTestDirectory for where to store failed tests
   // If not set then use the directory of the project's cypress.config where the test-results defaults to
-  const failedTestFilePath =
-    cypressConfig.env?.failedTestDirectory === undefined
-      ? `${appDir}/test-results/last-run.txt`
-      : `${cypressConfig.env.failedTestDirectory}/test-results/last-run.txt`;
+  const failedTestFilePath = `${appDir}/test-results/last-run.txt`;
```
[the remark this answers is not in the document]
Comment (dennisbergevin) [reply runFailed.js:14]: By using `const failedTestFilePath = ${appDir}/test-results/last-run.txt` as a solution, is your `cypress.config` file also in the same directory as `failedTestDirectory`?

- If yes, would it also be correct to say `If there are failed tests, run the following command from the directory of the project's cypress.config`?

- If no, how are your run commands/scripts detecting your config file if you are calling the `npx` script from a different directory than the `cypress.config`?


I can see this solution working if you do not set a `failedTestDirectory` and run the `npx` from the directory of the `cypress.config`, or simply set the `failedTestDirectory` to `./` when running from the directory of the `cypress.config`. 

I'm having trouble at the moment understanding how this works with `failedTestDirectory` being in a different directory than the `cypress.config`, would appreciate your insight!

[the remark this answers is not in the document]
Comment (DavidFontalba) [reply runFailed.js:14]: I understand your concern, but I believe the crucial point here is not the exact location of the `cypress.config` file, but rather ensuring that the path where the `.txt` file with the failed test results is generated matches the value defined in `failedTestFilePath`. To achieve this, it is necessary for the script to change the directory (`cd`) before being executed.

By using `const failedTestFilePath = ${appDir}/test-results/last-run.txt`, the file path is defined relative to the script's current directory. Therefore:

- If the `cypress.config` file is in the same directory as `failedTestDirectory`:
Then yes, it would be correct to say, "If there are failed tests, run the following command from the directory of the project's `cypress.config` file."

- If the `cypress.config` file is not in the same directory as `failedTestDirectory`:
In this case, for your commands/scripts to correctly detect your configuration file and generate the results in the desired path, you need to ensure the script changes to the appropriate directory before executing the Cypress commands. This can be done with a `cd` in the script before running the `npx` command as updated in the `README`.

In summary, the proposed solution will work if:
- You do not explicitly set `failedTestDirectory` or you set `failedTestDirectory` to `./` and run the `npx` command from the `cypress.config` directory.
- You explicitly set `failedTestDirectory` to a different path and you change your directory before executing the script.

[the remark this answers is not in the document]
Comment (dennisbergevin) [reply runFailed.js:14]: Thanks for the details and context, appreciate it, and I better see where you are coming from in your approach.

Could you give me an example of how you run the `npx` when `explicitly set failedTestDirectory to a different path and you change your directory before executing the script.`?

Are you also passing `--config-file` to the npm script you make so Cypress knows the path to the config?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dennisbergevin) [reply README.md:97]: Thanks for pointing this discrepancy out!

**Comment by dennisbergevin** [comment]
@DavidFontalba An alternative approach is if `test-results` simply is created and stored in the same directory as the cypress config, so no `failedTestDirectory` option.

This would simplify the logic and challenges, and if you gitignore the test results it appears to be more of an aesthetic locally on your repo.

Would love your thoughts on this alternative approach, again, really appreciate your feedback!

**Comment by DavidFontalba** [comment]
> @DavidFontalba An alternative approach is if `test-results` simply is created and stored in the same directory as the cypress config, so no `failedTestDirectory` option.
> 
> This would simplify the logic and challenges, and if you gitignore the test results it appears to be more of an aesthetic locally on your repo.
> 
> Would love your thoughts on this alternative approach, again, really appreciate your feedback!
> 
> I have a PR to address this here: #2

@dennisbergevin For me, that solution is as valid as any other and would solve our current problem in our TS Cypress project.
Thank you for the support and for publishing this handy plugin!

**Comment by dennisbergevin** [comment]
@DavidFontalba There is a new release 1.0.1 for the plugin including the updates, please feel free to reach out if you have any further issues!
